Apply linting suggestions to oscal-rest yaml files

tuckerzp opened the pull request:
This adds the ability to lint `.yaml` files as well as updates the `README` to describe those changes.

tuckerzp committed:
Start linting openapi.yaml

This adds a default yaml linter ruleset to run yamllint. openapi.yaml
failed the linter and has been updated.

tuckerzp committed:
Add Linting & Testing section to README

rgauss reviewed (commented):
One of the major concerns in encouraging/enforcing linting is that we need to conform to whatever Swagger Editor's YAML formatting defaults are.  From some quick tests, it looks like this still follows that app's format, so I think we're good.

@kylelaker has good comments as well.

Note that this does not need to block the actual open sourcing of this repository.

tuckerzp committed:
Add github workflow

This adds a very basic github workflow to lint yaml files upon a PR or
push to develop.

tuckerzp committed:
Add markdown linting

tuckerzp committed:
Use GitHub Super-Linter

laurelmay reviewed (changes requested):
This is good but a few quick requests:
 - I don't think that `.github/worfklows/pr.yml` needs all these comments; lots of them explain how GHA works. I think these come from their template? Let's yank those
 - `main` needs to be added to the list of branches to trigger on push.
 - need a comment explaining why that YAML Lint rule is disabled on line 6 (bc GH requires `on`)
 - can we specify `ubuntu-20.04` instead of `ubuntu-latest`? GHA does a very slow and wonky migration when a new LTS comes out. let's just stick with one version
 - please remove `workflow_dispatch` as a trigger (we don't want to trigger this manually)
 -

tuckerzp committed:
Clean up pr.yml

laurelmay reviewed (approved):
Thanks for cleaning all that up!
